src/spikeinterface/curation/remove_duplicated_spikes.py

```diff
@@ -39,48 +39,60 @@ def __init__(self, sorting: BaseSorting, censored_period_ms: float = 0.3, method
         seed = np.random.randint(low=0, high=np.iinfo(np.int32).max)
 
         for segment in sorting._sorting_segments:
-            self.add_sorting_segment(RemoveDuplicatedSpikesSortingSegment(segment, censored_period,
-                                                                          sorting.unit_ids, method, seed))
+            self.add_sorting_segment(
+                RemoveDuplicatedSpikesSortingSegment(segment, censored_period, sorting.unit_ids, method, seed)
+            )
 
         sorting.copy_metadata(self, only_main=False)
         if sorting.has_recording():
             self.register_recording(sorting._recording)
 
-        self._kwargs = {
-            'sorting': sorting,
-            'censored_period_ms': censored_period_ms,
-            'method': method
-        }
+        self._kwargs = {"sorting": sorting, "censored_period_ms": censored_period_ms, "method": method}
 
 
 class RemoveDuplicatedSpikesSortingSegment(BaseSortingSegment):
-
-    def __init__(self, parent_segment: BaseSortingSegment, censored_period: int, unit_ids, method: str, seed: Optional[int] = None) -> None:
+    def __init__(
+        self,
+        parent_segment: BaseSortingSegment,
+        censored_period: int,
+        unit_ids,
+        method: str,
+        seed: Optional[int] = None,
+    ):
         super().__init__()
         self._parent_segment = parent_segment
         self._duplicated_spikes = {}
+        self.censored_period = censored_period
+        self.seed = seed
+        self.method = method
 
-        for unit_id in unit_ids:
-            self._duplicated_spikes[unit_id] = \
-                find_duplicated_spikes(parent_segment.get_unit_spike_train(unit_id, start_frame=None, end_frame=None),
-                                       censored_period, method=method, seed=seed)
-
+        self._duplicated_spikes = {unit_id: None for unit_id in unit_ids}
 
-    def get_unit_spike_train(self, unit_id, start_frame: Optional[int] = None,
-                             end_frame: Optional[int] = None) -> np.ndarray:
+    def get_unit_spike_train(
+        self, unit_id, start_frame: Optional[int] = None, end_frame: Optional[int] = None
+    ) -> np.ndarray:
         spike_train = self._parent_segment.get_unit_spike_train(unit_id, start_frame=None, end_frame=None)
```

Comment:
Won't that be very heavy in the case of chunking?

Reply:
Good point. Is curation stuff passed to `ChunkRecordingExecutor`, we can use _kwargs to avoid that.